versioned_hdf5/tests/test_subchunk_map.py

```diff
@@ -409,6 +410,55 @@ def test_read_many_slices_param_nd(args):
     assert_array_equal(getitem_dst3, expect)
 
 
+@pytest.mark.slow
+@given(idx_shape_chunks_st(max_ndim=1))
+@hypothesis.settings(max_examples=max_examples, deadline=None)
+def test_entire_chunks_mapper(args):
+    idx, shape, chunks = args
+    _, mappers = index_chunk_mappers(idx, shape, chunks)
+    if not mappers:
+        return  # Early exit for empty index
+    assert len(shape) == len(chunks) == len(mappers) == 1
+    chunk_size = chunks[0]
+    orig_mapper = mappers[0]
+    entire_mapper = EntireChunksMapper(orig_mapper)
+
+    # Test chunks_indexer() and whole_chunks_idxidx()
+    all_chunks = np.arange(orig_mapper.n_chunks)
+    sel_chunks = all_chunks[orig_mapper.chunks_indexer()]
+    np.testing.assert_array_equal(
+        all_chunks[entire_mapper.chunks_indexer()], sel_chunks
+    )
+    np.testing.assert_array_equal(
+        sel_chunks[entire_mapper.whole_chunks_idxidx()], sel_chunks
+    )
+
+    # Test read_many_slices_params()
+    entire_mapper = EntireChunksMapper(orig_mapper)
```

Comment:
I think you can drop this, the state of `entire_mapper` has not changed, right?

Reply:
Good catch!